File: packages/ui_level3/tests/metadata.test.ts
import { describe, it, expect } from 'vitest'
import metadata from '../seed/metadata.json'

describe('UI Level 3 Metadata', () => {
  it('should have valid structure', () => {
    expect(metadata.packageId).toBe('ui_level3')
    expect(metadata.version).toMatch(/^\d+\.\d+\.\d+$/)
  })
  it('should depend on shared packages', () => {
    expect(metadata.dependencies).toContain('ui_permissions')
    expect(metadata.dependencies).toContain('ui_header')
  })
  it('should export level3 page', () => {
    expect(metadata.exports.pages).toContain('level3')
  })
})

File: packages/archive/screenshot_analyzer/seed/scripts/tests/metadata.test.lua
-- Metadata validation tests for screenshot_analyzer package
-- Uses lua_test framework

describe("Screenshot Analyzer Package Metadata", function()
  local metadata = load_cases("metadata.json")
  
  it("should have valid package structure", function()
    expect(metadata.name).toBe("screenshot_analyzer")
    expect(metadata.version).toBeTruthy()
    expect(metadata.description).toBeTruthy()
  end)
  
  it("should have semantic version", function()
    expect(metadata.version).toMatch("^%d+%.%d+%.%d+$")
  end)
  
  it("should have exports defined", function()
    expect(metadata.exports).toBeTruthy()
    expect(metadata.exports.components).toBeType("table")
  end)
  
  it("should have dependencies array", function()
    expect(metadata.dependencies).toBeType("table")
  end)
  
  it("should require minLevel 5", function()
    expect(metadata.minLevel).toBe(5)
  end)
end)
